integrations/azure_ai_search/src/haystack_integrations/document_stores/azure_ai_search/document_store.py

```diff
@@ -0,0 +1,450 @@
+
+        :param document_ids: ids of the documents to be deleted.
+        """
+        if self.count_documents == 0:
+            return
+        documents = self._get_raw_documents_by_id(document_ids)
+        if documents:
+            self.client.delete_documents(documents)
+
+    def get_documents_by_id(self, document_ids: List[str]) -> List[Document]:
+        return self._convert_search_result_to_documents(self._get_raw_documents_by_id(document_ids))
+
+    def search_documents(self, search_text: Optional[str] = "*", top_k: Optional[int] = 10) -> List[Document]:
+        """
+        Returns all documents that match the provided search_text.
+        If search_text is None, returns all documents.
+        :param search_text: the text to search for in the Document list.
+        :param top_k: Maximum number of documents to return.
+        :returns: A list of Documents that match the given search_text.
+        """
+        result = self.client.search(search_text=search_text, top=top_k)
+        return self._convert_search_result_to_documents(list(result))
+
+    def filter_documents(self, filters: Optional[Dict[str, Any]] = None) -> List[Document]:
+        """
+        Returns the documents that match the provided filters.
+        Filters should be given as a dictionary supporting filtering by metadata. For details on
+        filters, see the [metadata filtering documentation](https://docs.haystack.deepset.ai/docs/metadata-filtering).
+
+        :param filters: the filters to apply to the document list.
+        :returns: A list of Documents that match the given filters.
+        """
+        if filters:
+            normalized_filters = normalize_filters(filters)
+            result = self.client.search(filter=normalized_filters)
+            return self._convert_search_result_to_documents(result)
+        else:
+            return self.search_documents()
+
+    def _convert_search_result_to_documents(self, azure_docs: List[Dict[str, Any]]) -> List[Document]:
+        """
+        Converts Azure search results to Haystack Documents.
+        """
+        documents = []
+
+        for azure_doc in azure_docs:
+            embedding = azure_doc.get("embedding")
+            if embedding == self._dummy_vector:
+                embedding = None
+
+            # Anything besides default fields (id, content, and embedding) is considered metadata
+            meta = {
+                key: value
+                for key, value in azure_doc.items()
+                if key not in ["id", "content", "embedding"] and key in self._index_fields and value is not None
+            }
+
+            # Create the document with meta only if it's non-empty
+            doc = Document(
+                id=azure_doc["id"], content=azure_doc["content"], embedding=embedding, meta=meta if meta else {}
+            )
+
+            documents.append(doc)
+        return documents
+
+    def index_exists(self, index_name: Optional[str]) -> bool:
```

Comment:
internal method, no?

Reply:
For this one, I was wondering if it would be helpful to expose it? Can serve some purpose for user?

Reply:
I would avoid mentioning index, apart from initialization.
Some explanation here: https://github.com/deepset-ai/haystack-core-integrations/pull/1160#issue-2620805781

Reply:
Thanks for explaining. Agreed 👍